reframe/core/fields.py

```diff
@@ -100,23 +102,32 @@ def __set__(self, obj, value):
 
 
 class TimerField(TypedField):
-    '''Stores a timer in the form of a tuple ``(hh, mm, ss)``'''
+    '''Stores a timer in the form of a :class:`datetime.timedelta` object'''
 
     def __init__(self, fieldname, *other_types):
-        super().__init__(fieldname, types.Tuple[int, int, int], *other_types)
+        super().__init__(fieldname, datetime.timedelta, str,
+                         types.Tuple[int, int, int], *other_types)
 
     def __set__(self, obj, value):
         self._check_type(value)
-        if value is not None:
-            # Check also the values for minutes and seconds
+        if isinstance(value, tuple):
+            user_deprecation_warning(
+                'setting a timer field as tuple is deprecated: '
+                'please use a string <days>d<hours>h<minutes>m<seconds>s')
```

Comment:
```suggestion
                "please use a string in the following format '<days>d<hours>h<minutes>m<seconds>s'")
```

But also change the quotes above for consistency.